File: Sigmath.ConvertedToVBNet/engineering/kinematics.vb
Public Class Kinematics
	Public Shared Function CalculateDisplacement(initialVelocity As Double, finalVelocity As Double, acceleration As Double, time As Double) As Double
		Return ((finalVelocity + initialVelocity) / 2) * time + (0.5 * acceleration * Basic.Power(time, 2))
	End Function

	Public Shared Function CalculateFinalVelocity(initialVelocity As Double, acceleration As Double, displacement As Double) As Double
		Return Basic.SquareRoot(Basic.Power(initialVelocity, 2) + 2 * acceleration * displacement)
	End Function

	Public Shared Function CalculateTime(initialVelocity As Double, finalVelocity As Double, displacement As Double, acceleration As Double) As Double
		Return (2 * displacement) / (initialVelocity + finalVelocity)
	End Function

	Public Shared Function CalculateProjectileRange(initialVelocity As Double, launchAngle As Double) As Double
		Return (Basic.Power(initialVelocity, 2) * Trig.Sin(2 * launchAngle)) / Constants.G
	End Function



End Class

File: Sigmath.ConvertedToPython/engineering/kinematics.py
class Kinematics(object):
	def CalculateDisplacement(initialVelocity, finalVelocity, acceleration, time):
		return ((finalVelocity + initialVelocity) / 2) * time + (0.5 * acceleration * Basic.Power(time, 2))

	CalculateDisplacement = staticmethod(CalculateDisplacement)

	def CalculateFinalVelocity(initialVelocity, acceleration, displacement):
		return Basic.SquareRoot(Basic.Power(initialVelocity, 2) + 2 * acceleration * displacement)

	CalculateFinalVelocity = staticmethod(CalculateFinalVelocity)

	def CalculateTime(initialVelocity, finalVelocity, displacement, acceleration):
		return (2 * displacement) / (initialVelocity + finalVelocity)

	CalculateTime = staticmethod(CalculateTime)

	def CalculateProjectileRange(initialVelocity, launchAngle):
		return (Basic.Power(initialVelocity, 2) * Trig.Sin(2 * launchAngle)) / Constants.G

	CalculateProjectileRange = staticmethod(CalculateProjectileRange)